loadtest/user/userentity/bookmarks.go

```diff
@@ -0,0 +1,73 @@
+package userentity
+
+import (
+	"context"
+
+	"github.com/mattermost/mattermost/server/public/model"
+)
+
+// GetChannelBookmarks fetches bookmarks for the given channel since a specific timestamp.
+func (ue *UserEntity) GetChannelBookmarks(channelId string, since int64) error {
+	bookmarks, _, err := ue.client.ListChannelBookmarksForChannel(context.Background(), channelId, since)
+	if err != nil {
+		return err
+	}
+
+	return ue.store.SetChannelBookmarks(bookmarks)
+}
+
+// AddChannelBookmark creates a bookmark on the given channel
+func (ue *UserEntity) AddChannelBookmark(bookmark *model.ChannelBookmark) error {
+	bookmarkResp, _, err := ue.client.CreateChannelBookmark(context.Background(), bookmark)
+	if err != nil {
+		return err
+	}
+
+	return ue.store.AddChannelBookmark(bookmarkResp)
+}
+
+// UpdateChannelBookmark updates a given bookmark.
+func (ue *UserEntity) UpdateChannelBookmark(bookmark *model.ChannelBookmarkWithFileInfo) error {
+	patch := &model.ChannelBookmarkPatch{
+		FileId:      &bookmark.FileId,
+		DisplayName: &bookmark.DisplayName,
+		SortOrder:   &bookmark.SortOrder,
+		LinkUrl:     &bookmark.LinkUrl,
+		ImageUrl:    &bookmark.ImageUrl,
+		Emoji:       &bookmark.Emoji,
+	}
+
+	result, _, err := ue.client.UpdateChannelBookmark(context.Background(), bookmark.ChannelId, bookmark.Id, patch)
+	if err != nil {
+		return err
+	}
+
+	if result.Deleted != nil {
+		bId := result.Deleted.Id
+		if err := ue.store.DeleteChannelBookmark(bId); err != nil {
+			return err
+		}
+	}
```

Comment:
What situation leads to the returned bookmark being deleted?

Reply:
for example if the Id is an empty string or if a bookmark with that id is not really present in the memstore.. this should not happen but is to make sure the code is able to handle it.

Reply:
Got it, thanks for the clarification!